packages/ra-ui-materialui/src/list/filter/FilterListItem.tsx

```diff
@@ -144,27 +145,30 @@ const useStyles = makeStyles(theme => ({
  */
 const FilterListItem: FC<{ label: string; value: any }> = props => {
     const { label, value } = props;
-    const { filterValues, setFilters } = useListFilterContext();
+    const {
+        filterValues,
+        setFilters,
+        displayedFilters,
+        hideFilter,
+    } = useListContext();
     const translate = useTranslate();
     const classes = useStyles(props);
 
-    const isSelected = isEqual(value, filterValues);
+    const isSelected =
+        isEmpty(value) || isEmpty(filterValues)
+            ? false
+            : isMatch(filterValues, value);
```

Comment:
Problem: isEqual is wrong because we can have several filters in filterValues. We just want to know if "value" is part of filterValues.
Correction: using lodash isMatch instead